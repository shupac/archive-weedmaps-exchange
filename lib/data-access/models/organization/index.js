import { types } from 'mobx-state-tree';
import License from 'models/license';
import { Address } from 'models/address';
import Location from 'models/location';
import Brand from 'models/brand';

const Organization = types.model('OrganizationModel', {
  contactName: types.maybeNull(types.string),
  email: types.maybeNull(types.string),
  id: types.string,
  name: types.string,
  address: types.maybe(Address),
  organizationType: types.string,
  phoneNumber: types.maybeNull(types.string),
  licenses: types.array(License),
  description: types.maybeNull(types.string),
  brands: types.maybe(types.array(Brand)),
  locations: types.maybe(types.array(Location)),
});

export type OrganizationType = {
  contactName: string,
  id: string,
  email: string,
  name: string,
  organizationType: string,
  phoneNumber: string,
};

export default Organization;
