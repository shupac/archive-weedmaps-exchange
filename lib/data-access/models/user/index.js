import { types } from 'mobx-state-tree';
import Organization from 'models/organization';

export const WmProfile = types.model({
  username: '',
  email: '',
  slug: '',
  avatar_url: '',
});

const Preferences = types.model('Preferences', {
  locationId: types.maybeNull(types.string),
  userContext: types.string,
  brandId: types.maybeNull(types.string),
});

export const User = types.model({
  id: types.string,
  wmUserId: types.number,
  preferences: Preferences,
  email: types.string,
  organization: Organization,
});

export type UserType = {
  avatarUrl: string,
  wmUserId: string,
  username: string,
  userContext: string,
};
