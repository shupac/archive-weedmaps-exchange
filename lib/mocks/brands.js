export const mockBrands = [
  {
    slug: 'gfarmalabs',
    name: 'GfarmaLabs',
    description:
      '<p><strong><ins>Our Mission:</ins></strong></p> <p><em>Leader of the American Cannabis Revolution</em></p> <p> </p> <p>G FarmaLabs has become a well known brand throughout the cannabis community with both cannabis patients and connoisseurs alike. G FarmaLabs is committed to applying science and global resources to design and develop market leading brands in the cannabis flower, extracts, and edible product areas.</p> <p><strong><ins>History:</ins></strong></p> <p>Family owned and operated, G Farmalabs was conceived in 2007 by husband and wife team Ata and Nicole Gonzalez. Forced to relocate to California in 2009 after the financial crisis known as “The Great Recession,” Ata and Nicole believed the end of marijuana prohibition was near. Upon arriving in California, Ata attended Oaksterdam University and rented their first farm in Northern California. Together they carefully planted and harvested their first batch of marijuana flowers. Nicole is a member of a fourth generation tobacco family and has a strong background in the cigar industry. In 2013, G FarmaLabs launched America’s first complete brand of cannabis products.</p> <p>In 2015, Ata and Nicole’s idea had now become reality. G FarmaBrands was born. G FarmaLabs is the world’s first cannabis company that dedicates their effort to following the traditional lessons adapted from the Tobacco Industry. G FarmaLabs infused the two commercial enterprises to create a variety of products in an emerging market, setting the trend for the future of cannabis by merging the two industries.</p> <p><strong><ins>Process:</ins></strong></p> <p>G FarmaLabs works closely with a team of professionals who designed the California based processing center and production facility called the “Innovation Center” located in Desert Hot Springs, California. They have adopted the phrase “Napa Valley of Cannabis” and focus on blending the finest Domestic Cultivated Cannabis. Consistently cultivated, G FarmaLabs uses the same secret strains, each carefully chosen to guarantee a consistent experience.</p> <p>Perfecting the recipe and process, the G FarmaLabs farmers cultivate, harvest, dry, and trim their seven strains making it possible for customers to experience the aroma and feel the passion and precision the team at G FarmaLabs dedicates to their line of products.</p> <p><strong><ins>Products:</ins></strong></p> <p>Offering a variety of products from pre-rolls to edibles, G FarmaLabs caters to a wide market of cannabis users. G Stiks, their line of prestigious, 100% indoor grown pre-rolled joints are grown, processed, and rolled by G FarmaLabs and are available in 6 different options depending on user preference. For people who prefer Indica, G FarmaLabs G Stiks offers “ROJO”, packaged in red symbolizing a potent form of medicine. Patients in medical marijuana states enjoy the highly desired “VIOLETA” G Stik, packaged in purple, similar to violet, symbolizing various things from royalty to wealth, magic and mystery. The “AMARILLO” Sativa G Stiks symbolize a yellow butterfly purposely chosen to create a euphoric feeling for patients who suffer from depression. The “ROSADO” Sativa G Stik symbolizing one of the most celebrated flowers, the Pink Lotus Flower and is designed to uplift the heart and mind. The “VERDE” G Stik symbolizes energy and healing and the “NARAN-JA” G Stik symbolizes the Orange Rose. G FarmaLabs’ hybrid G Stiks offer a solution for connoisseurs on the go who need a product that is ready to go. The “VERDE” blend promotes energy healing and stress relief.</p> <p>In addition to G Stiks, G FarmaLabs also offers a line of vapor products, injectors, and edibles. Their Liquid Gold vape battery pens support 1.5 grams and are available in Super Silver Haze, Silver Haze, Blue Diamond, Green Crack , Orange Kush, Red Diesel, Pink Lemonade, Blackberry Kush, White Widow, and Purple Haze. They also proudly offer Indica, Sativa, and Hybrid Red Cannabis Oil Injectors in 0.5-1 gram dosages as well as NUG RUN Cannabis Oil injectors that are currently only offered in a 0.5 gram dosages designed to allow users to inhale Cannabinoids while avoiding harmful smoke toxins. G FarmaLabs offers infused decadent edible chocolate bars and truffles and 100% raw and unfiltered honey.</p> <p>From the detail of their packaging to their meticulous formula, G FarmaLabs proudly manufactures their own products delivering only the highest quality cannabis.</p> <p><strong><ins>Networks:</ins></strong></p> <p>Recognized as a Leader in the American Cannabis Revolution, G FarmaLabs has influenced celebrities such as The Game and has been featured on ABC News 7 and 790 KABC Talk Radio. G FarmaLabs can be found in dispensaries throughout Northern California, San Bernardino, Los Angeles, Orange County, Southern California, and Washington.</p>',
    id: 'd054b34f-428e-4583-a2a4-27f4e5aea6f3',
  },
  {
    slug: 'thclear-co',
    name: 'THClear Co',
    description:
      '<p>We are the industry leader in solventless distillate. We currently produce 6 different products with our Thclear Distillate, with our infamous uncut Private Reserve Cartridge being the most popular. Our other amazing products include: Disposable Pens, Cartridges, Syringes, Honeypot Vials, and our funky 50/50!</p>',
    id: 'd60ca06a-89db-4584-99a8-1f9b9f427e42',
  },
];
