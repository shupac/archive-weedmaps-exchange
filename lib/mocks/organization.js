export default {
  address: {
    territory: 'California',
    streetAddress: '41 Discovery',
    postalCode: '12345',
    longitude: 1,
    latitude: 2,
    country: 'USA',
    city: 'Irvine',
    id: '123456',
  },
  contactName: 'test contact',
  email: 'weedmaps@weedmaps.com',
  id: '123',
  licenses: [],
  name: 'test',
  organizationType: 'test org',
  phoneNumber: '1234566789',
}

export const mockOrgBrands = [
  {
    value: "9ffabab9-75bd-4d17-b8f6-265470243155",
    text: "BLVD Supply"
  },
  {
    value: "e98a5787-2e60-4302-b566-c6454a69a91f",
    text: "Harmony Extracts"
  },
  {
    value: "fd2b3edd-3de5-497b-af07-13d4cf02f240",
    text: "Bakked"
  },
  {
    value: "4853fd7e-8c28-4dac-84ae-9fde71d90d52",
    text: "Naked THC"
  },
  {
    value: "79577912-8b01-4194-b69d-e988a91d5113",
    text: "Natures Lab Extracts"
  },
  {
    value: "3072e38a-b3fb-4f0a-b5ef-bfe9bd6c0236",
    text: "Select"
  },
  {
    value: "6786ec8a-e3ea-4f4b-b23e-b55f3c9bff6d",
    text: "Flav"
  },
  {
    value: "16326c8a-0726-4a2b-92a4-57e5f67f97db",
    text: "West Coast Cure"
  },
  {
    value: "d60ca06a-89db-4584-99a8-1f9b9f427e42",
    text: "THClear Co"
  },
  {
    value: "d04dcd7c-933d-428e-a165-e91511371793",
    text: "Automation Test Brand CAsZyXnJvr"
  },
  {
    value: "d054b34f-428e-4583-a2a4-27f4e5aea6f3",
    text: "GfarmaLabs"
  }
]

export const mockOrg = {
  id: "2f4ddb82-3b67-4780-8ae4-7f27da36d813",
  phoneNumber: "1112223333",
  organizationType: "seller",
  name: "Andrew's Organization",
  email: "weedweed420@weed.com",
  description: "Update description",
  contactName: "Andrew Sciametta",
  licenses: [],
  brands: [
    {
      id: "9ffabab9-75bd-4d17-b8f6-265470243155",
      slug: "blvd-supply",
      shippingFee: 0,
      name: "BLVD Supply",
      minimumPurchasePrice: 0,
      description: "",
      deliveryEta: {
        etaMinUnit: "day",
        etaMin: 1,
        etaMaxUnit: "day",
        etaMax: 3
      }
    },
    {
      id: "e98a5787-2e60-4302-b566-c6454a69a91f",
      slug: "harmony-extracts",
      shippingFee: 0,
      name: "Harmony Extracts",
      minimumPurchasePrice: 0,
      description: "<p><ins><strong>Our Mission:</strong></ins></p> <p><em>Get Harmonized.</em></p> <p> </p> <p>Harmony Extracts passionately champions the cannabis and those within it. Wanting to do their part for the cannabis community, Harmony Extracts develops invariably excellent extract products in a long-time heart of the industry: Denver, Colorado. Here for everyone from the average smoker to the connoisseur market, Harmony Extracts assembles a team from the Old Guard of cannabis to bring the best quality and cleanest product to market at the most affordable prices. At Harmony Extracts, excellence is their norm.</p> <p><strong><ins>History:</ins></strong></p> <p>While Harmony Extracts officially started in August 2016, the passionate team behind the brand holds more than a decade’s worth of experience in the cannabis industry. Starting in 2003, the partners of Harmony Extracts began in cultivation before expanding into extraction and other leadership positions as the Colorado marijuana market became first medical and then recreational. As this team did everything from run cultivation centers to operate dispensaries, they soon felt a void in the marijuana market for consistent and quality extract product. The founders wanted to do things the right way, improving things from both a customer service and product standpoint and introducing a professionalism to the cannabis industry. Thus, Harmony Extracts sprung onto the scene fighting tooth and nail to meet Denver’s strict quality regulations and offer consistent, reliably excellent extractions.</p> <p>Today, Harmony Extracts lines the shelves of thirty dispensaries, where patients and recreational users alike notice their quality and recognize their brand. Harmony Extracts produces their world class products from their two state-of-the-art extraction facilities in Denver.</p> <p><strong><ins>Process:</ins></strong></p> <p>Hailing from many larger names in the industry, the extractors behind Harmony Extracts joined forces and combined the best methods to create their cannabis product. All of their flowers they grow in-house, or purchase from one or two of their exclusive vendors, to ensure their best materials remain the best. Drawing from classic strains and new, exciting genetics, Harmony Extracts uses butane and propane proprietary blend extraction to create the clearest and most potent shatter and wax. Their flagship product, their “Nectar” offers users an ultra-refined hydrocarbon extract with the potency of a distillate and the flavor of a live resin. Their proprietary method of extraction allows them to create a terpene-rich solution that crystallizes THCA, enabling them to separate the high-terpene cannabinoids from the crystals and reintroduce the plant materials at the end.</p> <p><strong><ins>Products:</ins></strong></p> <p>Harmony Extracts’ products started at “good” and have grown into “great” today.hey continually reevaluate their methods to keep delivering consistent excellence to their consumers. At the 2016 Hemp Connoisseur Championship, their four entries brought home eight different awards for Best Live Resin, Shatter and Wax. Always promising a high potency, their Nectar regularly tests at a potency of 99.99% THCA</p> <p><strong><ins>Lab Testing:</ins></strong></p> <p>Harmony Extracts exclusively tests their products with TEQ Analytical Laboratories in Aurora, Colorado, double checking their extracts for potency, terpene content, and any residual solvents.</p> <p><strong><ins>Networks:</ins></strong></p> <p>Currently, Harmony Extracts distributes their concentrate products to dispensaries throughout Colorado, found in popular dispensaries including Kind Love, Good Chemistry, Colorado Harvest Company and DANK in Denver, Boulder, and Crescent View. While they currently only work with recreational stores, they soon will offer products to medical patients. Harmony Extracts hopes to expand to more states by the start of 2018.</p>",
      deliveryEta: {
        etaMinUnit: "day",
        etaMin: 1,
        etaMaxUnit: "day",
        etaMax: 3
      },
      avatarImage: {
        id: "f8754a5a-01bd-46b3-ac72-5d2ae311ad28"
      }
    },
  ],
  address: {
    id: "4e0bb99c-0afb-4311-9cdc-8514d8f0a50c",
    territory: "CA",
    streetAddress: "1201 S Figueroa St",
    postalCode: "90015",
    longitude: -118.2688169,
    latitude: 34.0417609,
    country: "us",
    city: "Los Angeles"
  }
}