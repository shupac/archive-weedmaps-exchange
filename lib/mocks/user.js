
export const mockWmProfile = {
  username: "asciametta",
  avatar_url: "https://images-acceptance.weedmaps.com/users/002/208/463/avatar/square_fill/1544743547-image-1653344458.jpg",
  id: 2208463,
  email: "john+710@weedmaps.com",
  region_id: 478,
  name: "RESET TEST",
  slug: "asciametta",
  state: "",
  city: "",
  zip_code: "",
  banned_at: null,
  roles_mask: 1,
  roles: [
    "ops_manager"
  ],
  access: 0,
  confirmed_at: "2018-12-06T21:28:38.490Z",
  is_listing_owner: true,
  dob: "1988-03-12",
  marital_status: "none",
  gender: "none",
  iso_country_code: null,
  about_me: "",
  website_url: "",
  is_email_confirmed: true,
  facebook_id: "",
  instagram_id: "",
  twitter_id: "",
  large_image_url: "https://images-acceptance.weedmaps.com/users/002/208/463/avatar/large/1544743547-image-1653344458.jpg",
  reviews_count: 0,
  wayso_count: 2,
  following_count: 1,
  favorite_count: 6,
  accepts_newsletter: false
}

export const mockWmxUser = {
  id: "4612b157-1ee9-46e8-9d9b-314e3b9eb80c",
  wmUserId: 2208463,
  preferences: {
    userContext: "buyer",
    locationId: "6039ad85-7be7-45ce-a5f9-3e802eeba1e5"
  },
  email: "asciametta@weedmaps.com.eu",
  organization: {
    id: "2f4ddb82-3b67-4780-8ae4-7f27da36d813",
    phoneNumber: "1112223333",
    organizationType: "seller",
    name: "Some Org",
    email: "test@test.com",
    description: "Update description",
    contactName: "Paul Ruescher",
    locations: [],
    licenses: [],
    brands: [],
    address: {
      id: "4e0bb99c-0afb-4311-9cdc-8514d8f0a50c"
    }
  }
}

export const mockUser = {
  avatarUrl: "https://images-acceptance.weedmaps.com/users/002/208/463/avatar/square_fill/1544743547-image-1653344458.jpg",
  wmUserId: 2208463,
  username: "asciametta",
  userContext: "buyer"
}