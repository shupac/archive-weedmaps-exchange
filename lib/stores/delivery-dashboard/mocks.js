export const mockIndex = {
  data: {
    id: '1',
    type: 'index_card',
    relationships: {
      driver_cards: {
        data: [
          {
            type: 'driver_card',
            id: '1',
          },
          {
            type: 'driver_card',
            id: '2',
          },
          {
            type: 'driver_card',
            id: '3',
          },
        ],
      },
      route_cards: {
        data: [
          {
            type: 'route_card',
            id: '1',
          },
        ],
      },
      delivery_cards: {
        data: [
          {
            type: 'delivery_card',
            id: '1',
          },
          {
            type: 'delivery_card',
            id: '2',
          },
        ],
      },
    },
  },
  jsonapi: {
    version: '1.0',
  },
  included: [
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'started',
        destination_address: '1234',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'Alan Tirado',
        contact_email: 'atirado@weedmaps.com',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: '123-123-1234',
        item_count: 4,
        order_id: '1234',
        total: 1,
      },
      id: '1',
      relationships: {
        package: {
          data: {
            id: '1',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'created',
        destination_address: '11234 W S St',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'Bob Truman',
        contact_email: 'bob@truman.com',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: '343-345-3452',
        item_count: 4,
        order_id: '3421',
        total: 3455,
      },
      id: '2',
      relationships: {
        package: {
          data: {
            id: '1',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'started',
        destination_address: '11234 W S St',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'Bob Truman',
        contact_email: 'bob@truman.com',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: '343-345-3452',
        item_count: 4,
        order_id: '3421',
        total: 3455,
      },
      id: '3',
      relationships: {
        package: {
          data: {
            id: '3',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'assigned',
      },
      id: '1',
      relationships: {
        delivery_cards: {
          data: [
            {
              id: '1',
              type: 'delivery_card',
            },
          ],
        },
        driver: {
          data: {
            id: '1',
            type: 'driver_card',
          },
        },
      },
      type: 'route_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'in_progress',
      },
      id: '2',
      relationships: {
        delivery_cards: {
          data: [
            {
              id: '3',
              type: 'delivery_card',
            },
          ],
        },
        driver: {
          data: {
            id: '2',
            type: 'driver_card',
          },
        },
      },
      type: 'route_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'on_duty',
        name: 'Bob',
        phone_number: '714-277-9631',
        avatar_url:
          'https://images.weedmaps.com/users/002/120/830/avatar/square_fill/1510575803-1508943687-owl.png',
      },
      id: '1',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [
            {
              id: '1',
              type: 'route_card',
            },
          ],
        },
      },
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'on_duty',
        name: 'James',
        phone_number: '123-123-1234',
        avatar_url:
          'https://images.weedmaps.com/users/000/331/653/avatar/square_fill/1510567490-ShortRound.jpg?development=1',
      },
      id: '2',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [],
        },
      },
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'off_duty',
        name: 'Val',
        phone_number: '666-666-5656',
        avatar_url: null,
      },
      id: '3',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [],
        },
      },
    },
  ],
};
export const mockBaseIndex = {
  data: [
    {
      id: '1',
      type: 'index_card',
      relationships: {
        driver_cards: {
          data: [
            {
              type: 'driver_card',
              id: '1',
            },
          ],
        },
        route_cards: {
          data: [],
        },
        delivery_cards: {
          data: [
            {
              type: 'delivery_card',
              id: '1',
            },
          ],
        },
      },
    },
  ],
  jsonapi: {
    version: '1.0',
  },
  included: [
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'successful',
        destination_address: '1234',
        destination_latitude: 90,
        destination_longitude: 90,
        recipient_name: 'String',
        ordered_at: '2018-05-08T23:00:42.075253Z',
        contact_number: 'String',
        item_count: 4,
        order_id: 'String',
        total: 1,
      },
      id: '1',
      relationships: {
        package: {
          data: {
            id: '1',
            type: 'package',
          },
        },
      },
      type: 'delivery_card',
    },
    {
      attributes: {
        inserted_at: '2018-05-08T23:00:42.075253Z',
        status: 'free',
        name: 'Bob',
        phone_number: '714-277-9631',
        avatar_url:
          'https://images.weedmaps.com/users/002/120/830/avatar/square_fill/1510575803-1508943687-owl.png',
      },
      id: '1',
      type: 'driver_card',
      relationships: {
        in_progress_route_cards: {
          data: [],
        },
      },
    },
  ],
};
export const mockNewRoute = {
  data: {
    attributes: {
      inserted_at: '2018-05-08T23:00:42.075253Z',
      status: 'assigned',
    },
    id: '1',
    relationships: {
      delivery_cards: {
        data: [
          {
            id: '1',
            type: 'delivery_card',
          },
        ],
      },
      driver: {
        data: {
          id: '1',
          type: 'driver_card',
        },
      },
    },
  },
  included: [],
  jsonapi: { version: '1.0' },
};
export const mockNewDelivery = {
  data: {
    attributes: {
      inserted_at: '2018-05-08T23:00:42.075253Z',
      status: 'created',
      destination_address: '41 Discovery',
      destination_latitude: 33.666629,
      destination_longitude: -117.756304,
      recipient_name: 'Alan Tirado',
      ordered_at: '2018-05-08T23:00:42.075253Z',
      contact_number: '123-123-1234',
      item_count: 4,
      order_id: '420',
      total: 420,
    },
    id: '2',
    relationships: {
      package: {
        data: {
          id: '2',
          type: 'package',
        },
      },
    },
  },
  jsonapi: {
    version: '1.0',
  },
};
export const mockNewDriver = {
  data: {
    attributes: {
      inserted_at: '2018-05-08T23:00:42.075253Z',
      status: 'off_duty',
      name: 'Bob',
      phone_number: '123-123-1234',
      avatar_url:
        'https://images.weedmaps.com/users/002/120/830/avatar/square_fill/1510575803-1508943687-owl.png',
    },
    id: '3',
    type: 'driver_card',
    relationships: {
      in_progress_route_cards: {
        data: [],
      },
    },
  },
  jsonapi: {
    version: '1.0',
  },
};
export const mockDriverLocation = {
  data: {
    id: '1',
    type: 'driver_location',
    attributes: {
      latitude: 33.666629,
      longitude: -117.756304,
      occurred_at: '2018-05-08T23:00:42.075253Z',
    },
  },
  jsonapi: {
    version: '1.0',
  },
};
export const mockPackage = {
  data: {
    attributes: {
      contact_email: 'labron@weedmaps.com',
      contact_number: '555-555-5555',
      documents: [
        {
          image_url:
            'https://images-acceptance.internal-weedmaps.com/users/000/331/653/avatar/square_fill/1510567490-ShortRound.jpg',
          name: 'id',
        },
        {
          image_url:
            'https://images-acceptance.internal-weedmaps.com/users/002/306/378/avatar/square_fill/1531352592-download.jpeg',
          name: 'rec',
        },
      ],
      excise_tax: 1,
      inserted_at: '2018-05-08T23:00:42.075253Z',
      order_id: 'UUID-1234',
      order_items: [
        {
          name: 'Hardcore Og',
          price: 50,
          quantity: 1,
        },
      ],
      ordered_at: '2018-05-08T23:00:42.075253Z',
      recipient_name: 'Bob',
      recipient_note: 'Quick yo!',
      sales_tax: 1,
      subtotal: 1,
      total: 1,
      updated_at: '2018-05-08T23:00:42.075253Z',
    },
    id: '1234',
    type: 'package',
  },
  jsonapi: {
    version: '1.0',
  },
};
