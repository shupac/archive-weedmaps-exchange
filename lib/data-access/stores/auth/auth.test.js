import { getSnapshot } from 'mobx-state-tree';
import { mockWmProfile, mockWmxUser } from 'lib/mocks/user';
import { mockBrand } from 'lib/mocks/brands';
import { mockOrg } from 'lib/mocks/organization';
import AuthStore from './';

function setup() {
  const mockSdk = {
    request: {
      isAuthenticated: jest.fn().mockReturnValue(true),
    },
    user: {
      me: jest.fn().mockReturnValue(Promise.resolve(mockWmProfile)),
    },
  };

  const mockFetchClient = {
    put: jest.fn().mockReturnValue(),
    fetch: jest.fn().mockReturnValue(Promise.resolve({ data: mockWmxUser })),
  };

  return { mockSdk, mockFetchClient };
}

const org = {
  id: '2f4ddb82-3b67-4780-8ae4-7f27da36d813',
  name: 'Big Baller Brand',
  address: '1795 N 1st St, Hermiston, OR, US, 97838',
  contactName: 'Androoooooo',
  phoneNumber: '4442223332',
  email: 'andrew@andrew.com',
  licenses: [],
};

describe('Auth Store', () => {
  it('actions will update the state of the model', () => {
    const auth = AuthStore.create();
    auth.setUser(mockWmxUser);
    auth.setWmProfile(mockWmProfile);
    expect(getSnapshot(auth)).toMatchSnapshot();
  });

  it('fetches user data and sets user computed value', async () => {
    const { mockSdk, mockFetchClient } = setup();

    const auth = AuthStore.create(
      {},
      { wmSdk: mockSdk, client: mockFetchClient },
    );

    await auth.fetchUser();

    expect(auth.user).toEqual({
      avatarUrl:
        'https://images-acceptance.weedmaps.com/users/002/208/463/avatar/square_fill/1544743547-image-1653344458.jpg',
      userContext: 'buyer',
      username: 'asciametta',
      wmUserId: 2208463,
    });
    expect(auth.activeContext).toEqual('buyer');
    expect(auth.loggedIn).toBe(true);
  });

  it('should have computed value for orgBrands', async () => {
    const auth = AuthStore.create({ org: mockOrg });
    expect(await auth.orgBrands.length).toEqual(2);
  });

  it('should be able to setUserContext', async () => {
    const { mockSdk } = setup();
    const mockFetchClient = {
      put: jest.fn().mockReturnValue({ data: mockWmxUser }),
    };
    const auth = AuthStore.create(
      {},
      { wmSdk: mockSdk, client: mockFetchClient },
    );
    auth.setUserContext('buyer');

    expect(mockFetchClient.put).toHaveBeenCalledWith(`/users/me/preferences`, {
      data: {
        type: 'user_preferences',
        attributes: {
          user_context: 'buyer',
        },
      },
    });
  });

  it('should be able to fetchBrand', async () => {
    const mockFetchClient = {
      fetch: jest.fn().mockReturnValue(Promise.resolve({ data: mockBrand })),
    };

    const auth = AuthStore.create({}, { client: mockFetchClient });

    await auth.fetchBrand('d60ca06a-89db-4584-99a8-1f9b9f427e42');
    expect(auth.brand).toEqual({
      avatarImage: undefined,
      deliveryEta: {
        etaMax: 2,
        etaMaxUnit: 'week',
        etaMin: 1,
        etaMinUnit: 'hr',
      },
      description: undefined,
      id: 'd60ca06a-89db-4584-99a8-1f9b9f427e42',
      licenses: null,
      minimumPurchasePrice: 80,
      name: 'GFarmaLabs',
      shippingFee: 100,
      slug: undefined,
    });
  });

  it('should be able to updateBrand', async () => {
    const mockFetchClient = {
      put: jest.fn().mockReturnValue(),
    };

    const auth = AuthStore.create({}, { client: mockFetchClient });

    const brand = {
      minimum_purchase_price: 10,
      shipping_fee: 10,
      delivery_eta: {
        eta_min_unit: 'hr',
        eta_min: 1,
        eta_max_unit: 'week',
        eta_max: 2,
      },
      name: 'test',
      id: 'd60ca06a-89db-4584-99a8-1f9b9f427e42',
    };

    const response = await auth.updateBrand(brand);
    expect(mockFetchClient.put).toHaveBeenCalledWith(
      '/brands/d60ca06a-89db-4584-99a8-1f9b9f427e42',
      {
        data: {
          attributes: {
            delivery_eta: {
              eta_max: 2,
              eta_max_unit: 'week',
              eta_min: 1,
              eta_min_unit: 'hr',
            },
            id: 'd60ca06a-89db-4584-99a8-1f9b9f427e42',
            minimum_purchase_price: 10,
            name: 'test',
            shipping_fee: 10,
          },
          type: 'brand',
        },
      },
    );
    expect(response).toEqual(true);
  });

  it('should be able to updateOrganization', async () => {
    const mockFetchClient = {
      put: jest.fn().mockReturnValue(),
    };

    const auth = AuthStore.create({}, { client: mockFetchClient });

    const organization = {
      address: '123 Irvine',
      contactName: 'Test',
      email: 'weedmaps@weedmaps.com',
      name: 'test',
      phoneNumber: '123-123-1234',
      id: '1234',
    };

    await auth.updateOrganization(organization);
    expect(mockFetchClient.put).toHaveBeenCalledWith('/organizations/1234', {
      data: {
        attributes: {
          address: '123 Irvine',
          contact_name: 'Test',
          email: 'weedmaps@weedmaps.com',
          name: 'test',
          phone: '123-123-1234',
          undefined: '1234',
        },
        id: '1234',
        type: 'address',
      },
    });
  });

  it('should be able to setActiveBrand ', async () => {
    const mockFetchClient = {
      put: jest.fn().mockReturnValue({ data: mockWmxUser }),
    };
    const auth = AuthStore.create({}, { client: mockFetchClient });
    auth.setActiveBrand('9ffabab9-75bd-4d17-b8f6-265470243155');
    expect(mockFetchClient.put).toHaveBeenCalledWith(`/users/me/preferences`, {
      data: {
        type: 'user_preferences',
        attributes: {
          brand_id: '9ffabab9-75bd-4d17-b8f6-265470243155',
        },
      },
    });
  });

  it('handles getAuthStatus', async () => {
    const { mockSdk } = setup();
    const authStore = AuthStore.create({}, { wmSdk: mockSdk });
    expect(authStore.isAuthenticated).toEqual(true);
    mockSdk.request.isAuthenticated.mockReturnValue(false);

    expect(authStore.isAuthenticated).toEqual(false);
  });

  it('can update the users active location', async () => {
    const { mockSdk } = setup();
    const mockFetchClient = {
      put: jest.fn().mockReturnValue({ data: mockWmxUser }),
    };
    const auth = AuthStore.create(
      {},
      { wmSdk: mockSdk, client: mockFetchClient },
    );
    await auth.patchUserLocation('1234');

    expect(mockFetchClient.put).toHaveBeenCalledWith(`/users/me/preferences`, {
      data: {
        type: 'user_preferences',
        attributes: {
          location_id: '1234',
        },
      },
    });
  });

  it('can update the organization ', async () => {
    const mockFetchClient = {
      put: jest.fn().mockReturnValue({ data: mockOrg }),
    };

    const authStore = AuthStore.create({}, { client: mockFetchClient });
    const result = await authStore.updateOrganization(org);
    expect(result).toEqual(true);
  });

  it('should catch when updating the organization ', async () => {
    const mockFetchClient = {
      put: () => {
        throw new Error('test');
      },
    };

    const authStore = AuthStore.create({}, { client: mockFetchClient });
    const result = await authStore.updateOrganization(org);
    expect(result).toEqual(false);
  });
});
