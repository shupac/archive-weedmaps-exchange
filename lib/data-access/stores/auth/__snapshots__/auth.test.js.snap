// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Store actions will update the state of the model 1`] = `
Object {
  "brand": undefined,
  "org": undefined,
  "selectedLocation": "",
  "wmProfile": Object {
    "avatar_url": "https://images-acceptance.weedmaps.com/users/002/208/463/avatar/square_fill/1544743547-image-1653344458.jpg",
    "email": "john+710@weedmaps.com",
    "slug": "asciametta",
    "username": "asciametta",
  },
  "wmxUser": Object {
    "email": "asciametta@weedmaps.com.eu",
    "id": "4612b157-1ee9-46e8-9d9b-314e3b9eb80c",
    "organization": Object {
      "address": Object {
        "city": undefined,
        "country": undefined,
        "id": "4e0bb99c-0afb-4311-9cdc-8514d8f0a50c",
        "latitude": undefined,
        "longitude": undefined,
        "postalCode": undefined,
        "streetAddress": undefined,
        "territory": undefined,
      },
      "brands": Array [],
      "contactName": "Paul Ruescher",
      "description": "Update description",
      "email": "test@test.com",
      "id": "2f4ddb82-3b67-4780-8ae4-7f27da36d813",
      "licenses": Array [],
      "locations": Array [],
      "name": "Some Org",
      "organizationType": "seller",
      "phoneNumber": "1112223333",
    },
    "preferences": Object {
      "brandId": "fd2b3edd-3de5-497b-af07-13d4cf02f240",
      "locationId": "6039ad85-7be7-45ce-a5f9-3e802eeba1e5",
      "userContext": "buyer",
    },
    "wmUserId": 2208463,
  },
}
`;
